pkg/controller/stack/stack_controller.go

```diff
@@ -1126,35 +1134,28 @@ func (sess *reconcileStackSession) addDefaultPermalink(stack *pulumiv1.Stack) er
 		stack.Status.LastUpdate = &shared.StackUpdateState{}
 	}
 	stack.Status.LastUpdate.Permalink = shared.Permalink(info.URL)
-	err = sess.updateResourceStatus(stack)
+	err = sess.patchStatus(ctx, stack)
 	if err != nil {
-		sess.logger.Error(err, "Failed to update Stack status with default permalink", "Stack.Name", stack.Spec.Stack)
 		return err
 	}
 	sess.logger.Debug("Successfully updated Stack with default permalink", "Stack.Name", stack.Spec.Stack)
 	return nil
 }
 
-func (sess *reconcileStackSession) getLatestResource(o client.Object, namespacedName types.NamespacedName) error {
-	return sess.kubeClient.Get(context.TODO(), namespacedName, o)
-}
-
-func (sess *reconcileStackSession) updateResource(o client.Object) error {
-	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		return sess.kubeClient.Update(context.TODO(), o)
-	})
-}
-
-func (sess *reconcileStackSession) updateResourceStatus(o *pulumiv1.Stack) error {
-	name := types.NamespacedName{Name: o.Name, Namespace: o.Namespace}
-	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		var s pulumiv1.Stack
-		if err := sess.kubeClient.Get(context.TODO(), name, &s); err != nil {
-			return err
-		}
-		s.Status = o.Status
-		return sess.kubeClient.Status().Update(context.TODO(), &s)
-	})
+// patchStatus updates the recorded status of a stack using a patch. The patch is calculated with
```

Comment:
Thanks for this improvement! I think this was a source of frustration in the past due to conflicts.